src/Compilers/CSharp/Portable/CommandLine/CommandLineParser.cs

```diff
@@ -559,6 +560,10 @@ public new CSharpCommandLineArguments Parse(IEnumerable<string> args, string bas
                             optimize = false;
                             continue;
 
+                        case "deterministic":
```

Comment:
We should support deterministic+ and deterministic- as well.  See the optimize pattern above.